### PR TITLE
add skill-based matchmaking

### PR DESCRIPTION
**Pull request by cj-tomlin**
Issue #3

haven't tested, added a couple of commands for test

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -116,6 +108,22 @@ struct PlayerData {
     player_queueing_config: PlayerQueueingConfig,
 }
 
+impl Default for PlayerData {
```
Comment (Devon7925) [inline src/main.rs:111]: Forgot to do this, added it to my commit

File: src/main.rs [hunk 2]
```diff
@@ -34,6 +34,7 @@ struct QueueConfiguration {
     maps: Vec<String>,
     map_vote_count: u32,
     default_player_data: PlayerData,
+    skill_based_matchmaking: bool, // Indicates if skill-based matchmaking is enabled
```
Comment (Devon7925) [inline src/main.rs:37]: There is a way to remove skill based matchmaking under my system by modifying player config

File: src/main.rs [hunk 3]
```diff
@@ -157,7 +165,7 @@ async fn handler(
                             let mut player_data = data.player_data.lock().unwrap();
                             player_data
                                 .entry(new.user_id)
-                                .or_insert(config.default_player_data.clone()).queue_enter_time = None;
+                                .or_insert_with(PlayerData::default).queue_enter_time = None;
```
Comment (Devon7925) [inline src/main.rs:168]: It's actually intentional to not use default here: I want server staff to be able to modify default player data.

**Review by Devon7925 — COMMENTED** [review]
I was already far into working on this when I got your pr. My work is [here](https://github.com/Devon7925/queue_bot/commit/c334ac4c8a9b94250eef26b6fe1466ab0c4bfc7a). I appreciate you trying to help though.

File: src/main.rs (hunk 3 above)
Comment (cj-tomlin) [reply src/main.rs:168]: ah makes sense

**Pull request closed without merging**